### ISSUE TITLE
Error with yamal file in yolo11m-cls

### ISSUE THREAD

**Issue by DVMNtwo** [ultralytics/ultralytics#18261]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

I distributed the data for yolo11m-cls according to the example:

![image](https://github.com/user-attachments/assets/c35c56a9-f89a-4201-a8ba-0c68a46d4777)

This way each folder corresponds to the class of images in them.
But I don't know how to create a yaml file for this data form.

So how build yaml file for classification model?
I tried to use a yaml file for the detection model, but I got an error:

![image](https://github.com/user-attachments/assets/ec8e17df-9052-42ca-815e-b6a0092138cd)

**Comment by Y-T-G** [comment]
Classification doesn't use NMS. You just pass the path to folder as `data`.

https://docs.ultralytics.com/tasks/classify/#train

**Comment by DVMNtwo** [comment]
> Classification doesn't use yaml. You just pass the path to the dataset folder as `data`.
> 
> https://docs.ultralytics.com/tasks/classify/#train

Hello, @Y-T-G!
I have another problem, when I export yolo11m-clf and use its best.pt for inference I get the following error:

![image](https://github.com/user-attachments/assets/583fd5c5-58b5-4fa1-8edf-d2c59392ef5e)

I'm sure this file is 224 x 224, so why is this happening?

**Comment by UltralyticsAssistant** [comment]
A potential fix for this issue has been merged in [PR #22192: Verify passed dataset is a directory and not a file for classification training](https://github.com/ultralytics/ultralytics/pull/22192) by @Y-T-G with contributions from @glenn-jocher.

What changed and why this may resolve the issue:
- The classification dataset checker now validates that the provided path is a directory.
- If a file-like path is given (e.g., has a suffix), a clear ValueError is raised with guidance instead of attempting an auto-download.
- This avoids confusing behavior and helps ensure classification training uses the expected directory structure.
- See the [Classification dataset docs](https://docs.ultralytics.com/datasets/classify/) for the required layout.

How to test the fix:
- Install from the latest main branch:
  - `pip install -U "git+https://github.com/ultralytics/ultralytics.git@main"`
- Retry your workflow, ensuring the classification dataset path points to a directory (containing subfolders like /train and /val). Example:
  ```python
  from ultralytics import YOLO

  model = YOLO('yolo11n-cls.pt')
  model.train(data='path/to/classify_dataset')  # directory with /train and /val
  ```
- Alternatively, wait for the next official PyPI release and then:
  - `pip install -U ultralytics`

Feedback requested:
- Please let us know if the merged changes resolve the issue in your environment.
- If the problem persists, share a minimal snippet and the exact dataset path you’re passing (redact sensitive info).

Thanks for reporting this 🙏 and feel free to open follow-ups with any additional details.
